activation/poet.go

```diff
@@ -50,7 +50,9 @@ func checkRetry(ctx context.Context, resp *http.Response, err error) (bool, erro
 	if retry, err := retryablehttp.DefaultRetryPolicy(ctx, resp, err); retry || err != nil {
```

Comment:
this method will not fail with nil resp? does it make sense to exit immediately with false is resp is nil, just for additionnal defense?

Reply:
afaik `resp` is only `nil` if `err` is not, at least that's how `net/http` deals with failed requests.

That being said, we only want to overwrite the behaviour in the case of 404? Could we write this like this then:

```golang
func checkRetry(ctx context.Context, resp *http.Response, err error) (bool, error) {
	if resp != nil && resp.StatusCode == http.StatusNotFound {
		return true, err
	}
	return retryablehttp.DefaultRetryPolicy(ctx, resp, err)
}
```

Reply:
@fasmat, much nicer, thx. Updated the code.